cleanlab/internal/util.py

```diff
@@ -451,3 +451,20 @@ def compress_int_array(int_array, num_possible_values):
     if compressed_type is not None:
         int_array = int_array.astype(compressed_type)
     return int_array
+
+
+def smart_display_dataframe(df):
+    """Display a pandas dataframe if in a jupyter notebook, otherwise print it to console."""
+    try:
+        from IPython.display import display
+
+        # no exception, let's determine if we are running inside a jupyter notebook
+        shell = get_ipython().__class__.__name__
+        if shell == "ZMQInteractiveShell":
+            display(df)  # Jupyter notebook or qtconsole
+        elif shell == "TerminalInteractiveShell":
+            print(df)  # terminal, not notebook
+        else:
+            print(df)  # something else, but not notebook
```

Comment:
An alternative could be:  just `display(df)` in L460 and delete these extra checks. They anyway just goto L470, which would happen anyway if `try` clause fails.

I ask because not sure if `ZMQInteractiveShell` is overly limiting. Maybe `display(df)` works in other envs as well?  

Should keep as is if there's some specific situation where `display(df)` will not cause `try` to fail, but is not desirable to execute?

Reply:
FYI from my terminal console (shell  = TerminalInteractiveShell), both display(df) and print(df) work fine. They just give same output.

Reply:
Eg. in colab, I don't think `shell == "ZMQInteractiveShell"` but display(df) still would be preferred over print(df)

Reply:
Agree, but similarly do not know the answer to your last question. I can make this change, but it makes exception handling more challenging for developers in the future if we wish to expand this functionality.

Reply:
to your second and third comment, we still need error checking because in my console, i don't have IPython installed so it throws an error if i try to display